inference-engine/tests/ngraph_functions/src/utils/ngraph_helpers.cpp

```diff
@@ -200,29 +198,22 @@ void CompareNodes(const Node& actual, const Node& expected) {
 }  // namespace
 
 void CompareFunctions(const Function& actual, const Function& expected) {
-    const auto& actualResults = actual.get_results();
-    NGRAPH_CHECK(actualResults.size() == 1, "Got ", actualResults.size(), " outputs for function, but only single output functions are supported");
-    const auto& actualResult = actualResults.front();
-
-    const auto& expectedResults = expected.get_results();
-    NGRAPH_CHECK(expectedResults.size() == 1, "Got ", expectedResults.size(), " outputs for function, but only single output functions are supported");
-    const auto& expectedResult = expectedResults.front();
-
-    std::queue<ComparingNodesPair> nodes;
-    nodes.emplace(actualResult, expectedResult);
-    while (!nodes.empty()) {
-        const auto actualNode   = nodes.front().first;
-        const auto expectedNode = nodes.front().second;
-        nodes.pop();
-
-        CompareNodes(*actualNode, *expectedNode);
-
-        for (std::size_t i = 0; i < actualNode->inputs().size(); ++i) {
-            const auto& actualShape = actualNode->input(i).get_partial_shape();
-            const auto& expectedShape = expectedNode->input(i).get_partial_shape();
-            CompareShapes(actualShape, expectedShape);
+    const auto& actualOrderedOps = actual.get_ordered_ops();
+    const auto& expectedOrderedOps = actual.get_ordered_ops();
+
+    NGRAPH_CHECK(expectedOrderedOps.size() == actualOrderedOps.size(),
+                 "Functions compare: expected and actual ops number should be equal "
+                 "but got ", expectedOrderedOps.size(), " and ", actualOrderedOps.size(), " respectively");
 
-            nodes.emplace(actualNode->input_value(i).get_node_shared_ptr(), expectedNode->input_value(i).get_node_shared_ptr());
+    for (std::size_t i = 0; i < expectedOrderedOps.size(); i++) {
+        const auto& expectedOp = expectedOrderedOps[i];
+        const auto& actualOp = actualOrderedOps[i];
+
+        CompareNodes(*actualOp, *expectedOp);
+        for (std::size_t i = 0; i < actualOp->outputs().size(); ++i) {
```

Comment:
Since you changed direction of traverse - I guess in `CompareNodes` you should check outputs number instead of inputs.

BTW, I think we can check shapes and counts of inputs and outputs at the same time, just to be sure

Reply:
Agree, done.